docs: "What's new in Svelte" February newsletter (#10323)
sveltejs/svelte: documentation/blog/2024-02-01-whats-new-in-svelte-february-2024.md

---
title: "What's new in Svelte: February 2024"
description: 'New in Kit: `reroute`, `emulate` and more!'
author: Dani Sandoval
authorURL: https://dreamindani.com
---

Since SvelteKit 2.0 released late last year, there's been a bunch of new features to make the Kit dev experience even better! Meanwhile the Svelte team has been hard at work making Svelte 5 a reality.

In case you missed it, Rich also posted [Tenets](https://github.com/sveltejs/svelte/discussions/10085#discussion-6029409), "an attempt to articulate the Svelte philosophy — our bedrock principles, that guide our design decisions."

So let's dive in to check out what's new and see what the rest of the community has been up to...

## What's new in SvelteKit

> 

Reviewer: maybe mention the new `import { read } from '$app/server'` feature here, even though it's mentioned Learning Resources as well?

I think some people might gloss over Learning Resources and they are just interested in new features in SvelteKit/Svelte

Author: good catch! I'm not sure how I missed that 😅

In case you missed it, Rich also posted [Tenets](https://github.com/sveltejs/svelte/discussions/10085#discussion-6029409), "an attempt to articulate the Svelte philosophy — our bedrock principles, that guide our design decisions."

So let's dive in to check out what's new and see what the rest of the community has been up to...

## What's new in SvelteKit

- The client router is now tree shakeable (**2.1.0**, [#11340](https://github.com/sveltejs/kit/pull/11340))
- `$env/static/public` is now exposed in service workers (**2.2.0**, [Docs](https://kit.svelte.dev/docs/modules#env-static-public), [#10994](https://github.com/sveltejs/kit/pull/10994))
- `style-src-elem` is now supported in environments with a Content Security Policy (**2.2.1**, [Docs](https://kit.svelte.dev/docs/types#private-types-cspdirectives), [#11562](https://github.com/sveltejs/kit/pull/11562))
- The new `reroute` hook allows you to change how URLs are translated into routes (**2.3.0**, [Docs](https://kit.svelte.dev/docs/hooks#universal-hooks-reroute), [#11537](https://github.com/sveltejs/kit/pull/11537))

> - The `read` function is now available in the `$app/server` module - allowing you to read assets from filesystem (**2.4.0**, [Docs](https://kit.svelte.dev/docs/modules#app-server-read), [#11649](https://github.com/sveltejs/kit/pull/11649))

Reviewer 2: ```suggestion
- The `read` function is now available in the `$app/server` module - allowing you to read assets from the filesystem (**2.4.0**, [Docs](https://kit.svelte.dev/docs/modules#app-server-read), [#11649](https://github.com/sveltejs/kit/pull/11649))
```

> - Adapters can now provide an `emulate` for dev and preview servers. This is useful for providing access to e.g. KV namespaces in development (**2.5.0**, [Docs](https://kit.svelte.dev/docs/writing-adapters), [#11730](https://github.com/sveltejs/kit/pull/11730))

Reviewer 2: ```suggestion
- Adapters can now `emulate` the prod environment for dev and preview servers by implementing the corresponding functions. This is useful for providing access to e.g. KV namespaces in development (**2.5.0**, [Docs](https://kit.svelte.dev/docs/writing-adapters), [#11730](https://github.com/sveltejs/kit/pull/11730))
```

## What's new in Svelte

In case you missed it, [Svelte 5 is in preview](https://svelte-5-preview.vercel.app/docs/introduction). In the meantime, Svelte 4 (`@latest`) has had one bugfix release. There's just one new feature to highlight from the Svelte 5 changelog:

> - `$derived` props can now be modified (**5.0.0-next.31**, [Docs](https://svelte-5-preview.vercel.app/docs/runes#derived), [#10080](https://github.com/sveltejs/svelte/pull/10080))

Reviewer 2: ```suggestion
- Snippets can now take multiple arguments (**5.0.0-next.42**, [Docs](https://svelte-5-preview.vercel.app/docs/snippets), [#9988](https://github.com/sveltejs/svelte/pull/9988))
```
